src/main/java/org/folio/circulation/resources/CheckInByBarcodeResource.java

```diff
@@ -110,6 +112,8 @@ private void checkIn(RoutingContext routingContext) {
         processAdapter::getPickupServicePoint, CheckInContext::withHighestPriorityFulfillableRequest))
       .thenComposeAsync(updateItemResult -> updateItemResult.combineAfter(
         processAdapter::getRequester, CheckInContext::withHighestPriorityFulfillableRequest))
+      .thenComposeAsync(updateItemResult -> updateItemResult.combineAfter(
+        processAdapter::getDepartments, CheckInContext::withHighestPriorityFulfillableRequest))
```

Comment:
To avoid cluttering the main future-chain, fetching departments might be moved into `processAdapter::getRequester` above

Reply:
Removed it and now, fetching the departments after fetching the patrongroup in Userrepository